demos/janssen-tarp/browser-extension/src/features/authentication/components/AuthFlowInputs.tsx

```
@@ -121,13 +121,91 @@ export default function AuthFlowInputs({
+          chrome.tabs.onUpdated.addListener(listener);
+
+          // Handle user closing the popup manually
+          chrome.windows.onRemoved.addListener(function onClosed(winId) {
+            if (winId === popup.id) {
+              chrome.windows.onRemoved.removeListener(onClosed);
+              chrome.tabs.onUpdated.removeListener(listener);
+              reject(new Error('Authentication cancelled by user'));
+            }
+          });
+
+        })
```

Biome (2.4.14)

[error] Promise executor functions should not be `async`.

(lint/suspicious/noAsyncPromiseExecutor)